consensus/state.go

```diff
@@ -1012,19 +1306,17 @@ func (cs *ConsensusState) addVote(valIndex int, vote *types.Vote, peerKey string
 				}
 				if cs.Round <= vote.Round && prevotes.HasTwoThirdsAny() {
 					// Round-skip over to PrevoteWait or goto Precommit.
-					go func() {
-						cs.EnterNewRound(height, vote.Round, false)
-						if prevotes.HasTwoThirdsMajority() {
-							cs.EnterPrecommit(height, vote.Round, false)
-						} else {
-							cs.EnterPrevote(height, vote.Round, false)
-							cs.EnterPrevoteWait(height, vote.Round)
-						}
-					}()
+					cs.EnterNewRound(height, vote.Round) // if the vote is ahead of us
```

Comment:
I see, Enter* still has those round/step conditional filters.  Do you happen to know why we needed a new goroutine for these transitions earlier?